agent/command/git.go

```diff
@@ -298,7 +321,31 @@ func (c *gitFetchProject) buildSourceCloneCommand(ctx context.Context, comm clie
 		fmt.Sprintf("rm -rf %s", c.Directory),
 	}
 
-	cloneCmd, err := opts.getCloneCommand()
+	if !opts.fallbackToFullClone {
+		switch conf.Task.Requester {
+		case evergreen.GithubMergeRequester:
+			// If this is a github merge queue task, we can directly clone
+			// the provided branch.
+			if conf.GithubMergeData.HeadBranch != "" {
+				opts.branch = conf.GithubMergeData.HeadBranch
+			} else {
+				// If the head branch is missing, fall back to a full clone.
+				opts.fallbackToFullClone = true
+			}
+		case evergreen.GithubPRRequester:
+			// If this is a PR task, we can directly clone the provided PR.
+			if conf.GithubPatchData.HeadBranch != "" {
+				opts.branch = conf.GithubPatchData.HeadBranch
+				opts.owner = conf.GithubPatchData.HeadOwner
+				opts.repo = conf.GithubPatchData.HeadRepo
+			} else {
+				// If the head branch is missing, fall back to a full clone.
```

Comment:
Do we expect this case to be true, or are we just being careful? we could also probably just run a query on recent tasks to see if this is possible

Reply:
This case is true for old PR's (older than this commit). We could make a ticket to remove it, but it's also safe to keep in because if, for some reason, the HeadBranch is missing, the task will test on the wrong branch and will misinform the user of what ran (unless they look at agent logs and see what the exact git clone command)